### PR TITLE
netdata: 1.18.1 -> 1.19.0

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Update netdata to version 1.19.0.
Changelog - https://docs.netdata.cloud/changelog/#v1190-2019-11-27

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by Izorkin** [comment]
@GrahamcOfBorg test netdata

**Comment by Izorkin** [comment]
@GrahamcOfBorg test netdata

**Comment by Izorkin** [comment]
@GrahamcOfBorg build netdata

**Comment by tomberek** [comment]
Builds fine. Needs a few folders created for the default log,cache,lib to work. Even then there is an access permission problem when accessing the API/Web port. [NixOS]

**Comment by Izorkin** [comment]
@tomberek, if build package with the $ out / var folder, the service does not work:
```
netdata # [    9.578588] netdata[644]: SIGNAL: Not enabling reaper
netdata # [    9.581045] netdata[644]: 2019-12-08 07:56:56: netdata INFO  : MAIN : SIGNAL: Not enabling reaper
netdata # [    9.588671] netdata[644]: 2019-12-08 07:56:56: netdata ERROR : MAIN : Cannot open file '/nix/store/rjxximhk6dsxc21apchd9hqgxff3cbmq-netdata-1.19.0/var/log/netdata/debug.log'. Leaving 1 to its default. (errno 30, Read-only file system)
netdata # [    9.597320] netdata[644]: 2019-12-08 07:56:56: netdata ERROR : MAIN : Cannot open file '/nix/store/rjxximhk6dsxc21apchd9hqgxff3cbmq-netdata-1.19.0/var/log/netdata/error.log'. Leaving 2 to its default. (errno 30, Read-only file system)
netdata # [    9.605245] netdata[644]: 2019-12-08 07:56:56: netdata ERROR : MAIN : Cannot open file '/nix/store/rjxximhk6dsxc21apchd9hqgxff3cbmq-netdata-1.19.0/var/log/netdata/access.log'. Leaving -1 to its default. (errno 30, Read-only file system)
netdata # [    9.614230] netdata[644]: 2019-12-08 07:56:56: netdata INFO  : MAIN : resources control: allowed file descriptors: soft = 30000, max = 30000
netdata # [    9.622351] netdata[644]: 2019-12-08 07:56:56: netdata INFO  : MAIN : Adjusted my Out-Of-Memory (OOM) score from 0 to 1000.
netdata # [    9.625064] netdata[644]: 2019-12-08 07:56:56: netdata INFO  : MAIN : Adjusted netdata scheduling policy to idle (5), with priority 0.
netdata # [    9.631192] netdata[644]: 2019-12-08 07:56:56: netdata INFO  : MAIN : Running with process scheduling policy 'idle'
netdata # [    9.634213] netdata[644]: 2019-12-08 07:56:56: netdata ERROR : MAIN : Cannot chown '/run/netdata/netdata.pid' to 999:65534 (errno 1, Operation not permitted)
netdata # [    9.644229] netdata[644]: 2019-12-08 07:56:56: netdata ERROR : MAIN : Cannot fchown() fd 6. (errno 1, Operation not permitted)
netdata # [    9.646016] netdata[644]: 2019-12-08 07:56:56: netdata ERROR : MAIN : Cannot switch to user's netdata group (gid: 65534). (errno 1, Operation not permitted)
netdata # [    9.655189] netdata[644]: 2019-12-08 07:56:56: netdata ERROR : MAIN : Cannot become user 'netdata'. Continuing as we are.
netdata # [    9.656877] netdata[644]: 2019-12-08 07:56:56: netdata INFO  : MAIN : netdata started on pid 644.
netdata # [    9.663378] netdata[644]: 2019-12-08 07:56:56: netdata INFO  : MAIN : Executing /nix/store/rjxximhk6dsxc21apchd9hqgxff3cbmq-netdata-1.19.0/libexec/netdata/plugins.d/system-info.sh
netdata # [    9.666232] netdata[644]: Cannot open file '/nix/store/rjxximhk6dsxc21apchd9hqgxff3cbmq-netdata-1.19.0/var/log/netdata/debug.log'. Leaving 1 to its default.
netdata # [   10.126122] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_OS_NAME="NixOS"
netdata # [   10.130427] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_OS_ID=nixos
netdata # [   10.133507] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_OS_ID_LIKE=unknown
netdata # [   10.143246] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_OS_VERSION=20.03.git.134df38 (Markhor)
netdata # [   10.148264] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_OS_VERSION_ID=20.03.git.134df38
netdata # [   10.158486] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_OS_DETECTION=/etc/os-release[   10.259510] NET: Registered protocol family 17
netdata # [   10.172689] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_KERNEL_NAME=Linux
netdata # [   10.184082] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_KERNEL_VERSION=4.19.88
netdata # [   10.191499] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_ARCHITECTURE=x86_64
netdata # [   10.197345] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_VIRTUALIZATION=kvm
netdata # [   10.204113] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_VIRT_DETECTION=systemd-detect-virt
netdata # [   10.213080] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_CONTAINER=none
netdata # [   10.222301] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : NETDATA_SYSTEM_CONTAINER_DETECTION=systemd-detect-virt
netdata # [   10.226322] netdata[644]: 2019-12-08 07:56:57: netdata ERROR : MAIN : Cannot open the file /nix/store/rjxximhk6dsxc21apchd9hqgxff3cbmq-netdata-1.19.0/var/lib/netdata/health.silencers.json (errno 2, No such file or directory)
netdata # [   10.240281] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : CONFIG: cannot load user config '/nix/store/rjxximhk6dsxc21apchd9hqgxff3cbmq-netdata-1.19.0/etc/netdata/stream.conf'. Will try stock config.
netdata # [   10.249735] netdata[644]: 2019-12-08 07:56:57: netdata FATAL : MAIN :Cannot create unique machine id file '/nix/store/rjxximhk6dsxc21apchd9hqgxff3cbmq-netdata-1.19.0/var/lib/netdata/registry/netdata.public.unique.id'. Please fix this. # : Read-only file system
netdata # [   10.264026] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : /nix/store/rjxximhk6dsxc21apchd9hqgxff3cbmq-netdata-1.19.0/libexec/netdata/plugins.d/anonymous-statistics.sh 'FATAL' 'netdata:MAIN' '0321@registry/r:registry_get_th/30'
netdata: must succeed: curl --fail http://localhost:19999/
netdata # [   10.398022] netdata[644]: 2019-12-08 07:56:57: netdata ERROR : MAIN : child pid 676 exited with code 6.
netdata # [   10.402141] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : EXIT: netdata prepares to exit with code 1...
netdata # [   10.409292] netdata[644]: 2019-12-08 07:56:57: netdata INFO  : MAIN : /nix/store/rjxximhk6dsxc21apchd9hqgxff3cbmq-netdata-1.19.0/libexec/netdata/plugins.d/anonymous-statistics.sh 'EXIT' 'ERROR' '-'
```

**Comment by Izorkin** [comment]
@GrahamcOfBorg test netdata

**Comment by Izorkin** [comment]
Commit `nixosTests.netdata: add a sleep before testing` fixed this error
```
netdata: waiting for success: curl -s http://localhost:19999/api/v1/data\?chart=users.preads | jq -e '[.data[range(10)][.labels | indices("root")[0]]] | add | . > 0'
netdata # parse error: Invalid numeric literal at line 1, column 6
```

**Comment by Izorkin** [comment]
cc @Mic92

**Comment by Mic92** [comment]
Thanks!

File: nixos/tests/netdata.nix [hunk 1]
```diff
@@ -25,6 +25,7 @@ import ./make-test-python.nix ({ pkgs, ...} : {
 
     # check if the netdata main page loads.
     netdata.succeed("curl --fail http://localhost:19999/")
+    netdata.succeed("sleep 4")
```
Comment (Mic92) [inline nixos/tests/netdata.nix]: Why was this needed? Should this not be covered by `wait_until_succeeds`?

Comment (Izorkin) [reply nixos/tests/netdata.nix]: without `sleep 4` - error:
```
netdata: waiting for success: curl -s http://localhost:19999/api/v1/data\?chart=users.preads | jq -e '[.data[range(10)][.labels | indices("root")[0]]] | add | . > 0'
netdata # parse error: Invalid numeric literal at line 1, column 6
```

Comment (Izorkin) [reply nixos/tests/netdata.nix]: The netdata process cannot collect statistics `users.preads`
